### ISSUE TITLE
Migration failed on Metabase 0.30

### ISSUE THREAD

**Issue by alirezastack** [metabase/metabase#8288]
Current metabase version is `0.29.3` and I have downloaded the latest version `0.30`.

OS: Linux data-aggregator 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1 (2018-04-29) x86_64 GNU/Linux

When I run the metabase I give the exception below:

```
08-11 09:32:45 INFO metabase.util :: Loading Metabase...
08-11 09:32:49 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓
For more information, see https://www.metabase.com/docs/latest/operations-guide/start.html#encrypting-your-database-connection-details-at-rest
08-11 09:32:58 INFO metabase.core :: Starting Metabase in STANDALONE mode
08-11 09:32:58 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

08-11 09:32:58 INFO metabase.core :: Starting Metabase version v0.30.0 (04836dc release-0.30.0) ...
08-11 09:32:58 INFO metabase.core :: System timezone is 'Etc/UTC' ...
WARNING: any? already refers to: #'clojure.core/any? in namespace: monger.collection, being replaced by: #'monger.collection/any?
08-11 09:33:00 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
08-11 09:33:00 INFO metabase.db :: Verifying mysql Database Connection ...
08-11 09:33:01 INFO metabase.db :: Verify Database Connection ...  ✅
08-11 09:33:01 INFO metabase.db :: Running Database Migrations...
08-11 09:33:01 INFO metabase.db :: Setting up Liquibase...
08-11 09:33:01 INFO metabase.db :: Liquibase is ready.
08-11 09:33:01 INFO metabase.db :: Checking if Database has unrun migrations...
08-11 09:33:05 INFO metabase.db :: Database has unrun migrations. Waiting for migration lock to be cleared...
08-11 09:33:07 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@599df260: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:09 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@599df260: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:11 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@599df260: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:13 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@599df260: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:15 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@599df260: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:17 WARN metabase.util :: auto-retry metabase.db$run_schema_migrations_BANG_$fn__4851@5cc6f721: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:17 INFO metabase.db :: Setting up Liquibase...
08-11 09:33:17 INFO metabase.db :: Liquibase is ready.
08-11 09:33:17 INFO metabase.db :: Checking if Database has unrun migrations...
08-11 09:33:19 INFO metabase.db :: Database has unrun migrations. Waiting for migration lock to be cleared...
08-11 09:33:21 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@55eb10c0: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:23 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@55eb10c0: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:25 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@55eb10c0: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:27 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@55eb10c0: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:29 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775@55eb10c0: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
08-11 09:33:31 ERROR metabase.core :: Metabase Initialization FAILED
java.lang.Exception: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
	at metabase.db$wait_for_migration_lock_to_be_cleared$fn__4775.invoke(db.clj:159)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:394)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:398)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:398)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:398)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:398)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:398)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.db$wait_for_migration_lock_to_be_cleared.invokeStatic(db.clj:154)
	at metabase.db$wait_for_migration_lock_to_be_cleared.invoke(db.clj:150)
	at metabase.db$migrate_up_if_needed_BANG_.invokeStatic(db.clj:175)
	at metabase.db$migrate_up_if_needed_BANG_.invoke(db.clj:164)
	at metabase.db$migrate_BANG_$fn__4818.invoke(db.clj:274)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:747)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:717)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:782)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:717)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:730)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:717)
	at metabase.db$migrate_BANG_.invokeStatic(db.clj:262)
	at metabase.db$migrate_BANG_.invoke(db.clj:243)
	at metabase.db$run_schema_migrations_BANG_$fn__4851.invoke(db.clj:410)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:394)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:398)
	at metabase.util$do_with_auto_retries.invoke(util.clj:386)
	at metabase.db$run_schema_migrations_BANG_.invokeStatic(db.clj:409)
	at metabase.db$run_schema_migrations_BANG_.invoke(db.clj:392)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:428)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:421)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:161)
	at metabase.core$init_BANG_.invoke(core.clj:138)
	at metabase.core$start_normally.invokeStatic(core.clj:245)
	at metabase.core$start_normally.invoke(core.clj:239)
	at metabase.core$_main.invokeStatic(core.clj:265)
	at metabase.core$_main.doInvoke(core.clj:260)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
08-11 09:33:31 INFO metabase.core :: Metabase Shutting Down ...
08-11 09:33:31 INFO metabase.core :: Metabase Shutdown COMPLETE
```

I have issued the below command to release locks, but got no success:

`java -jar metabase.jar migrate release-locks`

Why is this happening on `Metabase`. I'm using the previous version `0.29.3` for now.

**Comment by jornh** [comment]
@alirezastack can you confirm this is with MySQL as Metabase application database? The log sort if hints to it.

**Comment by alirezastack** [comment]
@jornh sorry for incomplete data, here it is:

- Your databases: MySQL
- Metabase version: 0.30
- Metabase hosting environment: Linux
- Metabase internal database: MySQL

**Comment by alirezastack** [comment]
@senior before running the `release lock` command I issued the below command to make sure ENV variables for metabase is set:

```
# echo $MB_DB_TYPE
mysql

```

It seems that environment variables are set (other envs were set too). I did a clean migration again and hit to the previous error and I tried to release the lock but it was not successful (I can see the lock record with a value for `LOCKEDBY`):

```
mysql> select * from DATABASECHANGELOGLOCK where id=1;
+----+--------+---------------------+-------------------------+
| ID | LOCKED | LOCKGRANTED         | LOCKEDBY                |
+----+--------+---------------------+-------------------------+
|  1 |       | 2018-08-18 05:49:28 | 172.17.0.1 (172.17.0.1) |
+----+--------+---------------------+-------------------------+
1 row in set (0.00 sec)

```

And there was log file called `hs_err_pid22156.log` which is attached for further investigation.

Thank you for the help.

[hs_err_pid22156.log](https://github.com/metabase/metabase/files/2299185/hs_err_pid22156.log)

**Comment by alirezastack** [comment]
On migration I get the below error:

`08-18 10:50:07 WARN metabase.util :: auto-retry metabase.db$run_schema_migrations_BANG_$fn__4851@6083be44: Table 'QRTZ_JOB_DETAILS' already exists`

I cannot rollback to the previous version and it complains about fields that does not exist. What should I do now? Can I manually move questions and dashboards to the new metabase? I'm stuck now!

**Comment by senior** [comment]
@alirezastack The log you included looks like there's not enough memory available for even the minimum heap size which looks to be 256 MB.

From your next message, it seems like you've been able to fix the memory issue and clear the lock. You see a warning when 0.29.3 starts up, is it functional otherwise? We warn if you start an older version of Metabase on a new Metabase database because usually it is an accident, but in your case it's intentional. We try to structure our migrations such that older versions will still run on the newer databases for situations like this. That doesn't always work, but we do try. Does your 0.29.3 version still work other than the warning?

Regardless of whether 0.29.3 is working or not, we'll still need to get your database fixed before you're able to upgrade to 0.30.0. What seems to have happened is you were partially through running migration 89, but didn't finish it and somehow some of the tables of the migration were created, but not all of them. That migration is creating entirely new tables that can all be safely dropped. When Metabase 0.30.0 starts up and finds that migration 89 hasn't ran, it'll run it and create the tables again.

To fix this up you'll need to run some SQL directly against your Metabase database. First verify that the database changelog does NOT include a row for changeset `89`. This query should return no rows:
```
select * from DATABASECHANGELOG where id='89';
``` 
Next we will manually delete the new tables created by that changeset. We don't know when the migration failed and there's not a built-in way with MySQL to only delete FK constraints when they're present, so run this SQL below, but it will probably fail because not all of the constraints will exist:
```
ALTER TABLE QRTZ_TRIGGERS
DROP FOREIGN KEY FK_QRTZ_TRIGGERS_JOB_DETAILS;
ALTER TABLE QRTZ_SIMPLE_TRIGGERS
DROP FOREIGN KEY FK_QRTZ_SIMPLE_TRIGGERS_TRIGGERS;
ALTER TABLE QRTZ_CRON_TRIGGERS
DROP FOREIGN KEY FK_QRTZ_CRON_TRIGGERS_TRIGGERS;
ALTER TABLE QRTZ_SIMPROP_TRIGGERS
DROP FOREIGN KEY FK_QRTZ_SIMPROP_TRIGGERS_TRIGGERS;
ALTER TABLE QRTZ_BLOB_TRIGGERS
DROP FOREIGN KEY FK_QRTZ_BLOB_TRIGGERS_TRIGGERS;
```
With the FK constraints gone, we can drop the tables, this should not fail:
```
DROP TABLE IF EXISTS QRTZ_CALENDARS;
DROP TABLE IF EXISTS QRTZ_PAUSED_TRIGGER_GRPS;
DROP TABLE IF EXISTS QRTZ_FIRED_TRIGGERS;
DROP TABLE IF EXISTS QRTZ_SCHEDULER_STATE;
DROP TABLE IF EXISTS QRTZ_LOCKS;
DROP TABLE IF EXISTS QRTZ_BLOB_TRIGGERS;
DROP TABLE IF EXISTS QRTZ_SIMPROP_TRIGGERS;
DROP TABLE IF EXISTS QRTZ_CRON_TRIGGERS;
DROP TABLE IF EXISTS QRTZ_TRIGGERS;
DROP TABLE IF EXISTS QRTZ_SIMPLE_TRIGGERS;
DROP TABLE IF EXISTS QRTZ_JOB_DETAILS;
```
Now when you startup 0.30.0, it should recreate those tables and you should get past the `'QRTZ_JOB_DETAILS' already exists` error

**Comment by alirezastack** [comment]
I removed the table `QRTZ_JOB_DETAILS` and hit another error because of utf8mb4 charset (changed it to utf8).
Everything works fine now. Thank you.
